### ISSUE TITLE
Use of `Assembly.DefinedTypes` makes the process brittle

### ISSUE THREAD

**Issue by SteveGilham** [ejball/XmlDocMarkdown#88]
It's all to easy to start processing an assembly and ending up with a ResolutionLoadFailure.

Cases so far encountered that can cause type resolution failures

1) dependency of the .net framework assembly of interest on a .net .exe assembly even if co-located
2) dependency of the .net framework assembly of interest on a system assembly that doesn't get copied to the build output location e.g references to "Microsoft.Build.Framework"and "Microsoft.Build.Utilities.v4.0"
3) when the .net core/standard assembly has been built, but not published, so dependencies are not co-located, but remain in the .nuget cache or other standard locations.

These can be worked around by copying the appropriate files (cases 1 and 2), or using a .net framework target instead for case 3.  Cases 1 and 3, however, are susceptible to the use of a ResolveEventHandler.

**Comment by SteveGilham** [comment]
Expands upon issue #69 (which covers case 3 only)

**Comment by SteveGilham** [comment]
Proof of concept of the `ResolveEventHandler` approach [here](https://github.com/SteveGilham/XmlDocMarkdown/tree/develop/issue-88); doesn't have any of the administrivia yet to make a PR.

**Comment by ejball** [comment]
I recently got a "could not load file or assembly" error on `System.Text.Json` from a GitHub Actions build for no obvious reason. I still don't know why it happened, but I ended up working around it by [building a custom console app](https://github.com/ejball/XmlDocMarkdown/issues/69#issuecomment-686763869) to generate the docs. It's frustrating not to be able to use `xmldocmd` directly but I'm hoping it will be a good long-term workaround.
